quic/s2n-quic-tests/src/tests/drop_initial_packets.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+use s2n_quic::provider::{
+    endpoint_limits::{ConnectionAttempt, Limiter, Outcome},
+    tls::default::{self as tls},
+};
+use s2n_quic_core::{connection::error::Error, endpoint};
+
+/// A custom limiter that always returns Outcome::close()
+struct AlwaysCloseLimiter;
+
+impl Limiter for AlwaysCloseLimiter {
+    fn on_connection_attempt(&mut self, _info: &ConnectionAttempt) -> Outcome {
+        Outcome::close()
+    }
+}
+
+// This test verifies that when the server would send a CONNECTION_CLOSE frame with
+// error code CONNECTION_REFUSED when the server's limiter returns Outcome::close().
+#[test]
+fn drop_initial_packets_test() {
+    let model = Model::default();
+
+    let connection_close_subscriber = recorder::ConnectionClosed::new();
+    let connection_close_event = connection_close_subscriber.events();
+
+    test(model, |handle| {
+        let server_tls_setup = tls::Server::builder()
+            .with_application_protocols(["h3"].iter())
+            .unwrap()
+            .with_certificate(certificates::CERT_PEM, certificates::KEY_PEM)
+            .unwrap()
+            .build()
+            .unwrap();
```

Comment:
do you need this? can you just do `.with_tls(SERVER_CERTS)` in the server builder?